### ISSUE TITLE
Worker crashes upon restart due to broken linked list

### ISSUE THREAD

**Issue by haerdib** [Polkadex-Substrate/polkadexTEE-worker#79]
The worker crashed upon restart due to `Linkedlist is broken` in function [verify_pdex_account_read_proofs(..)](https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/develop/enclave/src/polkadex.rs#L58-L61).

This happens in the following scenario:
1) start node
2) start worker
3) register main account //Alice with `bin/substratee-client register-account //Alice` (currently only available on branch  [ocex-client-calls](https://github.com/Polkadex-Substrate/polkadexTEE-worker/tree/ocex-client-calls))
4) restart worker

This crash is caused due to `account.account.prev != last_account`. 
@Gauthamastro @ksr30 before I look into it, do you perhaps have any input on what goes wrong there?

**Comment by Gauthamastro** [comment]
It's a bug, https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/develop/enclave/src/polkadex.rs#L50

L50 is a wrong statement I believe, it should be like 
```rust
if account.account.next.is_some() {           
         last_account = account.account.current.clone().unwrap();                
} else {

break;
}


```

**Comment by jaxter03** [comment]
@haerdib should we close it? ... I think it's done

**Comment by haerdib** [comment]
Hm. I just added your commit to the openfinex feature branch to be able to test the openfinex server. But it's not yet on the develop branch. Could you do a PR from your bug fix branch? the openfinex branch is not yet ready for a PR.

**Comment by haerdib** [comment]
Or we can just leave it open and close it with the PR from the openfinex branch. That's also works.

**Comment by jaxter03** [comment]
No problem @haerdib ... I will raise new PR.

**Comment by haerdib** [comment]
Closed with #95
